components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -2493,3 +2512,215 @@ def object_table_query(request, objtype, objid, conn=None, **kwargs):
+        #     initial = {'server': unicode(server_id)}
+        #     form = self.form_class(initial=initial)
+        # else:
+        #     form = self.form_class()
+        return {"message": "POST only with username, password and csrftoken"}
+
+    def _handleLoggedIn(self, request, conn, connector, *args, **kwargs):
+        """ Returns a response for successful login """
+        c = conn.getEventContext()
+        ctx = {}
+        for a in ['sessionId', 'sessionUuid', 'userId', 'userName', 'groupId',
+                  'groupName', 'isAdmin', 'eventId', 'eventType',
+                  'memberOfGroups', 'leaderOfGroups']:
+            if (hasattr(c, a)):
+                ctx[a] = getattr(c, a)
+        return {"success": True, "eventContext": ctx}
+
+    def _handleNotLoggedIn(self, request, error, **kwargs):
+        """ Returns a response for failed login """
+        # Since @jsonp decorator can't return a 403,
+        # we do it manually. NB: this won't return jsonp 'callback()'
+        return JsonResponseForbidden({"message": error})
+
+    def post(self, request, *args, **kwargs):
+        error = None
+        form = self.form_class(request.POST.copy())
+        useragent = 'OMERO.webgateway_api'
+        if form.is_valid():
+            username = form.cleaned_data['username']
+            password = form.cleaned_data['password']
+            server_id = form.cleaned_data['server']
+            is_secure = form.cleaned_data['ssl']
+
+            connector = Connector(server_id, is_secure)
+
+            # TODO: version check should be done on the low level, see #5983
+            compatible = True
+            if settings.CHECK_VERSION:
+                compatible = connector.check_version(useragent)
+            if (server_id is not None and username is not None and
+                    password is not None and compatible):
+                conn = connector.create_connection(
+                    useragent, username, password,
+                    userip=get_client_ip(request))
+                # TODO: conn is None if user is INACTIVE (not in user group)...
+                if conn is not None:
+                    # Check if user is in "user" group
+                    roles = conn.getAdminService().getSecurityRoles()
+                    userGroupId = roles.userGroupId
+                    # ... so this will ALWAYS be True
+                    if userGroupId in conn.getEventContext().memberOfGroups:
+                        request.session['connector'] = connector
+                        # UpgradeCheck URL should be loaded from the server or
+                        # loaded omero.web.upgrades.url allows to customize web
+                        # only
+                        try:
+                            upgrades_url = settings.UPGRADES_URL
+                        except:
+                            upgrades_url = conn.getUpgradesUrl()
+                        upgradeCheck(url=upgrades_url)
```

Comment:
could you pass alternative agent like OMERO.webapi that we will know who is using it?
We need new agent in qa db

cc: @sbesson @joshmoore @jburel